hydra-node/test/Hydra/PersistenceSpec.hs

```diff
@@ -60,20 +60,17 @@ spec = do
       monadicIO $ do
         items <- pick $ listOf genPersistenceItem
         moreItems <- pick $ listOf genPersistenceItem `suchThat` ((> 2) . length)
-        result :: Either PersistenceException () <- run $ withTempDir "hydra-persistence" $ \tmpDir -> do
-          persistence@PersistenceIncremental{loadAll, append} <- createPersistenceIncremental $ tmpDir <> "/data"
-          forM_ items append
-          loadAll `shouldReturn` items
-          try $ loadAndAppendConcurrently persistence moreItems
-
-        monitor $ counterexample $ show result
-        assert $ isLeft result
-
-loadAndAppendConcurrently :: PersistenceIncremental Value IO -> [Value] -> IO ()
-loadAndAppendConcurrently PersistenceIncremental{loadAll, append} moreItems =
-  race_
-    (forever $ threadDelay 0.01 >> loadAll)
-    (forM_ moreItems $ \item -> append item >> threadDelay 0.01)
+        pure $
+          withTempDir "hydra-persistence" $ \tmpDir -> do
+            PersistenceIncremental{loadAll, append} <- createPersistenceIncremental $ tmpDir <> "/data"
+            forM_ items append
+            loadAll `shouldReturn` items
+            race_
+              (forever $ threadDelay 0.01 >> loadAll)
+              (forM_ moreItems $ \item -> append item >> threadDelay 0.01)
+              `shouldThrow` \case
+                IncorrectAccessException{} -> True
```

Comment:
Funnily, I had something like that initially, but I found giving a proper name to the `race_ ...` block more legible and stating explicitly what it's supposed to do instead of requiring the user to unpack the whole thing.